src/base/bittorrent/session.cpp

```diff
@@ -4376,13 +4376,17 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
 
 void Session::handleStateUpdateAlert(const lt::state_update_alert *p)
 {
+    QVector<BitTorrent::TorrentHandle *> updatedTorrents;
```

Comment:
Apparently, you are guided by some point of libtorrent documentation, which says that there may not be all torrents? Please point it out to me.

Reply:
>Apparently, you are guided by some point of libtorrent documentation, which says that there may not be all torrents? 

Did you really checked the doc before asking?
https://www.libtorrent.org/reference-Alerts.html#state-update-alert
>It contains the torrent status of all torrents that changed since last time this message was posted.

Reply:
>Did you really checked the doc before asking?

Of course not (not now), otherwise I wouldn't ask. I wonder why we haven't tried the optimization before? Or tried, but it didn't work as we expected?..

Reply:
no idea, I wasn't paying attention to it until now.